### PR TITLE
New graph implementation

### PR DESCRIPTION
**Pull request by muhuk**
### This patch includes:

- petgraph based module hierarchy & dependency implementation.
- Visitor implementation that populates the graph.
  - Populating the graph and printing are separate steps.
- Implementation of `tree` command, with `--plain` & `--orphans` support.

### Next step:

- `graph` command implementation with edition 2018 support.

### Notes:

- All new files are under `ng` module.  Once this is done we can move them to `src/` and remove `ng`.
- It works exactly as before unless `--enable-edition-2018` is specified.  If `graph` command is passed with this flag on, it prints:

  ```shell
  Warning: Edition 2018 support is unstable.

  graph is not implemented for Edition 2018 yet.
  Try removing --enable-edition-2018
  ```

I would like to merge this now as it has become a bit large.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -21,7 +21,9 @@ repository = "https://github.com/regexident/cargo-modules"
 version = "0.4.4"
 
 [dependencies]
-colored = "1.7.0"
+arrayvec = "0.4.10"
```
Comment (muhuk) [inline Cargo.toml]: This is required for `ng::graph::Module`.

File: src/main.rs [hunk 2]
```diff
@@ -54,6 +61,42 @@ fn choose_target<'a>(args: &Arguments, manifest: &'a Manifest) -> Result<&'a Tar
...
+        .map(|t| t.src_path().clone())
+        .collect();
+    let target: &Target = choose_target(args, &manifest)?;
+    let include_orphans: bool = args.orphans;
+    let enable_color: bool = if cfg!(target_os = "windows") {
+        false
+    } else {
+        !args.plain
+    };
+    colored::control::set_override(enable_color);
```
Comment (muhuk) [inline src/main.rs]: Once we do this, we don't need to keep doing `if color_enabled {...`

File: test-resources/project/test-lib/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+[package]
```
Comment (muhuk) [inline test-resources/project/test-lib/Cargo.toml]: I had added this originally, but then I didn't write tests against it.  Better clean it up.

File: src/main.rs (hunk 2 above)
Comment (regexident) [reply src/main.rs]: That is indeed much nicer!

**Comment by muhuk** [comment]
Looks like @hmvp is busy with other stuff.  I will merge this in the weekend, we can fix any post-merge comments in the next patch.

**Comment by regexident** [comment]
>Looks like @hmvp is busy with other stuff.  I will merge this in the weekend, we can fix any post-merge comments in the next patch.

@muhuk sounds good to me!

**Comment by muhuk** [comment]
Thanks.  Merging.

**Pull request merged**